providers/azure.go

```diff
@@ -149,13 +157,40 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (*
 	created := time.Now()
 	expires := time.Unix(jsonResponse.ExpiresOn, 0)
 
-	return &sessions.SessionState{
+	session := &sessions.SessionState{
 		AccessToken:  jsonResponse.AccessToken,
 		IDToken:      jsonResponse.IDToken,
 		CreatedAt:    &created,
 		ExpiresOn:    &expires,
 		RefreshToken: jsonResponse.RefreshToken,
-	}, nil
+	}
+
+	// the presence of id_token depends on the configuration and
+	// verification of id_token requires oidc configuration
+	if session.IDToken != "" && p.Verifier != nil {
+		email, err := p.verifyIDTokenAndExtractEmail(ctx, session.IDToken)
+		if err != nil {
+			return nil, err
+		}
+		session.Email = email
+	}
+
+	return session, nil
+}
+
+func (p *AzureProvider) verifyIDTokenAndExtractEmail(ctx context.Context, rawIDToken string) (string, error) {
+	idToken, err := p.Verifier.Verify(ctx, rawIDToken)
+	if err != nil {
+		return "", err
+	}
+	var idTokenClaims struct {
```

Comment:
`ProviderData` has `getClaims` available if that helps here

Reply:
sounds good